### ISSUE TITLE
Uniforme continuité utilisee sans justification page 656

### ISSUE THREAD

**Issue by Provatiscus** [LaurentClaessens/mazhe#124]
Page 656 Candidat limite.

On dit que f est continue sur Q et que par conséquent pour tout p et q dans Q on a |  p-q  |< d --> |f(p) - f(q)| < epsilon. On a donc utilisé une notion de continuité uniforme sans justification

**Comment by LaurentClaessens** [comment]
Bien vu.
À mon avis, il faut fixer x et travailler sur un compact autour de x. Alors f est continue sur ce compact et donc uniformément continue.

Par contre, comme cette preuve est bourrée de chausse-trappe de topologies différentes et induites, il faut faire très attention avant d'agir à la légère. Je vais y réfléchir.

**Comment by Provatiscus** [comment]
Merci. D'ailleurs le même problème apparaît plusieurs fois dans la preuve du ou des thm qui suivent.
J'ai eu la même réaction que toi. Avec un compactage ça devrait marcher.... sauf que Q n'est pas localement compact donc ça pose problème

**Comment by LaurentClaessens** [comment]
Ah ouais, c'est un peu chaud.
Ce serait pas mal de prouver que si K est compact dans R, alors (K inter Q) est compact dans Q.

**Comment by LaurentClaessens** [comment]
Je ne suis pas d'accord avec le fait que l'union des B' recouvre K.
En effet, prenons un irrationnel r dans K.
Pour chaque q, je peux prendre d_q un peu plus petit pour que r ne soit pas dans B(q, d_q).

On peut même faire ça très fort. Si (r_k) est une suite dont la série vaut M, alors si (q_i) est une énumération des rationnels dans R, 
B(q_i, r_i) donne une partie ouverte de R qui contient tous les rationnels et dont la mesure est M.

**Comment by LaurentClaessens** [comment]
Je crois que la proposition 18.124 (dans la section 18.16 Prolongement de fonctions) fait ce qu'on veut.

**Comment by Provatiscus** [comment]
Oui tu as raison ! Merci !

**Comment by LaurentClaessens** [comment]
Non, je me plante, 18.124 ne fait pas le boulot; elle est pour les fonctions linéaires.

**Comment by LaurentClaessens** [comment]
Et il y a pire. À mon avis la proposition est même fausse; l'hypothèse de continuité n'a pas l'air de suffire.
https://fr.wikipedia.org/wiki/Fonction_Cauchy-continue

**Comment by Provatiscus** [comment]
Je disais tu as raison pour le contre exemple ^^

**Comment by LaurentClaessens** [comment]
Après réflexion, la principale difficulté restante est de montrer que la fonction

q-> q^(1/n)
est Cauchy-continue. C'est le principal cas d'utilisation du lemme de prolongation de Q vers R, pour définir la fonction puissance.

**Comment by LaurentClaessens** [comment]
Pour la fonction puissance, peut-être qu'il faut suivre ceci :
http://irma.math.unistra.fr/~bopp/CAPES/cours/equation-felle-exp.pdf

Elle ne définit pas la puissance via une limite, mais en disant que a^x est le sup des a^q avec q<x. Ensuite, elle prouve que le tout est continue et comme il faut.

**Comment by LaurentClaessens** [comment]
Ok. La situation est résolue.
- ajout de l'hypothèse de Cauchy-continuité dans le lemme incriminé
- mise à jour de la preuve
- démonstration du fait que la fonction q-> a^q est Cauchy-continue.
- je garde la définition de a^x par la prolongation continue de Q vers R

Pour moi c'est bon.

**Comment by Provatiscus** [comment]
Super merci !
